doc/tutorials/namespaces.md

```diff
@@ -1,27 +1,93 @@
-# Namespaces #
+# Understanding Namespaces #
 
-## Definition ##
+## Structure of the key database ##
 
-In Elektra multiple keys may refer to the same part of the configuration.
-A _namespace_ is the part of the key that refers to the key's origin (like a unique path).
+The _key database_ of Elektra is _hierarchically structured_. This means that keys are organized similar to directories in a file system.
 
-Possible sources are: 
+Lets add some keys to the database. To add a key we can use this command:
 
-- **spec/something** for specification of other keys
-- **proc/something** for in-memory keys (e.g. command-line)
-- **dir/something** for dir keys in current working directory
-- **system/something** for system keys in /etc or /
-- **user/something** for user keys in home directory
-- **/something** for cascading keys (actually refers to one of the above)
+	kdb set <key> <value>
 
-Having namespaces enables both admins and users to set specific parts of the application's configuration.
+Now add the the key **/a** with the Value **Value 1** and the key **/b/c** with the Value **Value 2**:
+
+	kdb set /a 'Value 1'
+	kdb set /b/c 'Value 2'
+
+![alt text](img/elektra_hierarchy.svg "Hierarchical structure of key database")
```

Comment:
I think the markdown parser can handle all syntaxes (not sure about the reference image syntax, but we don't use it anyway). The problem could be one of my scripts that normalizes links and image paths in the documents before they get copied to the website directory. I think at the moment it only works with `![Alt text](/path/to/img.jpg)`, but not with the other syntax. This is because I'm doing only one replace for the syntax `[text](path)`, which applies to both links and images. I'm not so sure how to know whether something is a link, but no image...?

In my opinion, the alt text can be used for the description above (if it is only something short). The alt text is the hover-text, where the optional description is meant for more detailed explanations. But I think there is no standard that defines what is done with the description (where it is placed in the rendered markdown).